gnosis-buyback/contracts/Buyback.sol

```diff
@@ -0,0 +1,756 @@
+    );
+
+    event ModifyExternalPoke(
+        address indexed userAddress,
+        bool allowExternalPoke
+    );
+
+    /**
+     * @notice Create the buyback contract
+     * @param _dx Address of the dutch exchange
+     */
+    function BuyBack(address _dx) public {
+        require(address(_dx) != address(0));
+        dx = DutchExchange(_dx);
+        owner = msg.sender;
+    }
+       
+    /**
+     * @notice addBuyBack
+     * @param _userAddress Address of the user 
+     * @param _buyToken Address of the buy token
+     * @param _sellToken Address of the sell token
+     * @param _burn Burn the token after buy back success
+     * @param _auctionIndexes Auction indexes the to participate 
+     * @param _auctionAmounts Amounts to participate in each realtive auction index
+     * @param _timeInterval Minimum time passed between buyback execution
+     * @param _allowExternalPoke Allow non owners to call postSellOrder
+    */
+    function addBuyBack(
+        address _userAddress,
+        address _buyToken,
+        address _sellToken, 
+        address _burnAddress, 
+        bool _burn, 
+        uint[] _auctionIndexes, 
+        uint[] _auctionAmounts,
+        uint _timeInterval,
+        bool _allowExternalPoke,
+        uint tipAmount
+        ) public {
+
+        require(address(_userAddress) != address(0));
+        require(address(_buyToken) != address(0));
+        require(address(_sellToken) != address(0));
+        require(_sellToken != _buyToken);
+        require(_auctionIndexes.length == _auctionAmounts.length);
+        require(buybacks[_userAddress].sellToken == address(0)); // ensure the user address doesn't exist
+        
+        // // map the auction ids to the auction amount
+        for(uint i = 0; i < _auctionIndexes.length; i++){
+            require(_auctionAmounts[i] > 0); // ensure the auction amount is greater than zero
+            auctionIndexWithAmount[_userAddress][_auctionIndexes[i]] = _auctionAmounts[i];
+        }
+
+        Buyback memory buyback = Buyback(
+            _sellToken, 
+            _buyToken, 
+            _burnAddress,
+            _burn, 
+            _auctionIndexes, 
+            _timeInterval, 0, true, 
+            _allowExternalPoke );
+
+        // map user address to the buyback config
+        buybacks[_userAddress] = buyback;
+        
+        // set tip price
+        tips[_userAddress] = tipAmount; 
+
+        emit AddBuyBack(
+            _userAddress,
+            _buyToken,
+            _sellToken, 
+            _burnAddress, 
+            _burn, 
+            _auctionIndexes, 
+            _auctionAmounts
+        );
+    }
+
+    /**
+     * @notice modifyAuctionAmountMulti Modify the amount for multiple auction index
+     * @param _userAddress User addresss
+     * @param _auctionIndexes Auction index the to participate 
+     * @param _auctionAmounts Auction amount to fill in auction index
+     */
+    function modifyAuctionAmountMulti(
```

Comment:
Yeah, I figured a project can want to increase or decrease the buyback amount they want to execute based off parameters external to the contract.

Take for example Binance coin (BNB), Binance executes buyback of BNB based off the profits for the month so enabling the project to modify it accordingly seemed okay.

Reply:
The buy back must be enforced, otherwise loosed it's usability. It should be an unbreakable commitment. 

If you want to modify, it should be to allow to add new buybacks, but users should have the certainty that they'll participate in the auction.